packages/kit/src/exports/vite/index.js

```diff
@@ -516,12 +516,17 @@ export function set_building() {
 							input,
 							output: {
 								format: 'esm',
-								// we use .mjs for client-side modules, because this signals to Chrome (when it
-								// reads the <link rel="preload">) that it should parse the file as a module
-								// rather than as a script, preventing a double parse. Ideally we'd just use
-								// modulepreload, but Safari prevents that
-								entryFileNames: ssr ? '[name].js' : `${prefix}/[name].[hash].mjs`,
-								chunkFileNames: ssr ? 'chunks/[name].js' : `${prefix}/chunks/[name].[hash].mjs`,
+								// see the kit.output.preloadStrategy option for details on why we have multiple options here
+								entryFileNames: ssr
+									? '[name].js'
+									: `${prefix}/[name].[hash]${
+											kit.output.preloadStrategy === 'preload-mjs' ? '.mjs' : '.js'
+									  }`,
```

Comment:
Can we DRY out this and the section below?

```suggestion
									: `${prefix}/[name].[hash].${ext}`,
```